kornia/utils/image.py

```diff
@@ -4,57 +4,111 @@
 import torch
 
 
+def hw_image_to_hw_tensor(image: np.array) -> torch.Tensor:
+    if len(image.shape) != 2:
+        raise ValueError("Input size must be a two dimensional array")
+    tensor: torch.Tensor = torch.from_numpy(image)
+    return tensor
+
+
+def hwc_image_to_chw_tensor(image: np.array) -> torch.Tensor:
+    if len(image.shape) != 3:
+        raise ValueError("Input size must be a three dimensional array")
+    tensor: torch.Tensor = torch.from_numpy(image)
+    tensor = tensor.permute(2, 0, 1)
+    return tensor
+
+
+def bhwc_image_to_bchw_tensor(image: np.array) -> torch.Tensor:
+    if len(image.shape) != 4:
+        raise ValueError("Input size must be a four dimensional array")
+    tensor: torch.Tensor = torch.from_numpy(image)
+    tensor = tensor.permute(0, 3, 1, 2)
+    return tensor
+
+
 def image_to_tensor(image: np.array) -> torch.Tensor:
     """Converts a numpy image to a PyTorch tensor image.
 
     Args:
-        image (numpy.ndarray): image of the form :math:`(H, W, C)`.
+        image (numpy.ndarray): image of the form :math:`(H, W)`, math:`(H, W, C)`, 
+        or math:`(B, H, W, C)`.
 
     Returns:
-        torch.Tensor: tensor of the form :math:`(C, H, W)`.
+        torch.Tensor: tensor of the form :math:`(H, W)`, math:`(C, H, W)`, 
+        or math:`(B, C, H, W)`.
 
     """
-    if not type(image) == np.ndarray:
+    if not isinstance(image, np.ndarray):
         raise TypeError("Input type is not a numpy.ndarray. Got {}".format(
             type(image)))
 
-    if len(image.shape) > 3 or len(image.shape) < 2:
-        raise ValueError("Input size must be a two or three dimensional array")
+    if len(image.shape) > 4 or len(image.shape) < 2:
+        raise ValueError(
+            "Input size must be a two, three or four dimensional array")
+
```

Comment:
I would keep this much more simple:
```python
tensor: torch.Tensor = torch.from_numpy(image)
if len(input_shape) > 2:
    tensor = tensor.transpose(-3, -2).transpose(-2, -1)
```
In this case we can handle (*,C,H,W) and (H,W).
Similar will be for `tensor_to_image`

Reply:
The transpose approach seems not work.
```
>>> a = torch.ones([2,4,4,3])
>>> b = a.transpose(-3, -2).transpose(-2, -1)
>>> b.shape
torch.Size([2, 4, 3, 4])
```
In the new commit, I moved extra functions, and move all the transpose to if-elif statement.

Reply:
sorry the indices were wrong, try this: 

```python
a = torch.ones(2, 4, 5, 3)
b = a.transpose(-1, -2).transpose(-2, -3)
c = b.transpose(-3, -2).transpose(-2, -1)
assert a.shape == c.shape
```

Reply:
@jiangwei221 have you tried ?

Reply:
hi, sorry for the late reply. 
yes, the new order works perfectly. 
however, I think use if-elif-else + permute is easier to understand, and to maintain: the implementation for different shape is seperated, and the permute order for each shape is intuitive, and there would be only one operation be applied on torch tensor which is "permute".
what do you think?

Reply:
I'm not a fan of if-elif-else codes since don't scale at all. In this case what if a user wants to convert a batch of video or an input with more dimensions ? we'll have to modify each time a user find a new use case and in case the library doesn't satisfy his needs, we'll most-likely loose a user. Unless we get some numbers I don't think this operation is so critical in terms of performance. As for the code readability, if you feel like is not clear, with a good side comment it could be :D